src/Nethermind/Nethermind.Serialization.Json/EthereumJsonSerializer.cs

```diff
@@ -143,18 +143,15 @@ private JsonWriterOptions CreateWriterOptions(bool indented)
         public async ValueTask<long> SerializeAsync<T>(Stream stream, T value, bool indented = false, bool leaveOpen = true)
         {
             var writer = GetPipeWriter(stream, leaveOpen);
-            Serialize(writer, value, indented);
+            await JsonSerializer.SerializeAsync(writer, value, indented ? JsonOptionsIndented : _jsonOptions);
```

Comment:
Does it make sense to be `ValueTask` method now? Won't that always allocate?

Reply:
`SerializeAsync` returns a `Task`, so it will always allocate?

Reply:
Then this method will allocate an additional Task vs if it synchronously completes it won't?

Reply:
won't it always allocate as the inner Task is not synchronous?

Reply:
The inner task will normally be synchrounous; will only go async when the outbound http buffer is full and it suspends

Reply:
Why didn't they make it ValueTask then?